src/tables/math.rs

```diff
@@ -0,0 +1,1062 @@
+//! A [MATH Table](https://docs.microsoft.com/en-us/typography/opentype/spec/math) implementation.
+
+use crate::GlyphId;
+use crate::gpos::Device;
+use crate::opentype_layout::Coverage;
+use crate::parser::{
+    Offset16, Stream,
+    FromSlice, FromData, Offset,
+    LazyArray16, LazyOffsetArray16,
+};
+
+/// A [MATH Table](https://docs.microsoft.com/en-us/typography/opentype/spec/math).
+#[allow(missing_debug_implementations)]
+#[allow(missing_docs)]
+#[derive(Clone)]
+pub struct Table<'a> {
+    pub major_version: u8,
+    pub minor_version: u8,
+    pub math_constants: MathConstants<'a>,
+    pub math_glyph_info: MathGlyphInfo<'a>,
+    pub math_variants: MathVariants<'a>,
+}
+
+#[derive(Debug, Default, Copy, Clone)]
+struct RawMathValueRecord {
+    value: i16,
+    device_table_offset: Option<Offset16>,
+}
+
+impl FromData for RawMathValueRecord {
+    const SIZE: usize = 4;
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        Some(RawMathValueRecord {
+            value: s.read()?,
+            device_table_offset: s.read()?,
+        })
+    }
+}
+
+impl RawMathValueRecord {
+    fn get(self, data: &[u8]) -> MathValueRecord {
+        let device_table = self.device_table_offset
+            .and_then(|offset| data.get(offset.to_usize()..))
+            .and_then(Device::parse);
+        MathValueRecord { value: self.value, device_table }
+    }
+}
+
+/// A [MathValueRecord](https://docs.microsoft.com/en-us/typography/opentype/spec/math#mathvaluerecord).
+#[allow(missing_debug_implementations)]
+#[allow(missing_docs)]
+#[derive(Default, Debug, Copy, Clone)]
```

Comment:
No need to implement `Default` unless necessary.

Reply:
This `Default` implementation is used later for `unwrap_or_default` in accessors of `MathConstants`.